cmd/spire-server/cli/federation/create_test.go

```diff
@@ -153,48 +138,58 @@ func TestCreate(t *testing.T) {
 		fakeResp  *trustdomainv1.BatchCreateFederationRelationshipResponse
 		serverErr error
 
-		expOut string
-		expErr string
+		expOutPretty string
+		expOutJSON   string
+		expErrPretty string
+		expErrJSON   string
 	}{
 		{
-			name:   "Missing trust domain",
-			expErr: "Error: trust domain is required\n",
+			name:         "Missing trust domain",
+			expErrPretty: "Error: trust domain is required\n",
+			expErrJSON:   "Error: trust domain is required\n",
```

Comment:
So, I've identified two scenarios in which we can have two different behaviors for printing the result when some error occurs:
- An error can occur before we make an API call, and in that case, the error message will be the same for `pretty` or `json` formats, one example is the test case that you highlighted here.
- An error can occur after the API call, on the server side, and it can come described in the API response. In this case, it wouldn't make sense to print the same error message as `pretty` format,  since we have a structured API response, and this error indication is contained in it, so for `json` format option, it makes more sense to continue representing the server response with json. You can find a test case example on line 311 (Federation relationships that failed to be created are printed).

Reply:
But if that is the case, are not we returning inconsistent errors? |
since in some cases (in this example only one) we'll be able get text but in another ones we'll get a json

I think we must be consistent, and if we expect json return json. or just return always text

Reply:
Another way to see this discussion is by expanding the formatting options with YAML format, let's say we are running a command for `list agent`, and we want the output to be formatted in YAML format. If some error occurs while making the request to the server API, and the response could not be retrieved, I wouldn't expect the error output of that command to be returned as a YAML, but rather I would expect a text format, and that same logic would apply for JSON as well

Reply:
We've brought this subject to discussion in our last contributor sync; taking into account the examples provided here and the fact that we have services that include RPCs with multiple response statuses (in batch), we have agreed on the following:
- Print errors in text format
- Print the API response in the requested format, regardless of whether it includes a message with [status ](https://github.com/spiffe/spire-api-sdk/blob/main/proto/spire/api/types/status.proto#L5) different than OK.

Reply:
👍 thanks @guilhermocc 

I may add the further clarification that, any API response for which the request response code is OK, the payload of the response should be printed in the format requested, even if the response payload itself includes gRPC codes other than OK.

Any API response code that is not OK, or any error encountered while issuing the request or reading the response, should be printed in pretty print format.